pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -135,77 +135,23 @@ def restore_model_state(self, model: LightningModule, checkpoint) -> None:
         # restore model state_dict
         model.load_state_dict(checkpoint['state_dict'])
 
-    def restore_training_state(self, checkpoint, load_optimizer_states: bool = True):
+    def restore_training_state(self, checkpoint: Dict[str, Any]) -> None:
         """
-        Restore trainer state.
-        Model will get its change to update
-        :param checkpoint:
-        :return:
+        Restore the trainer state from the pre-loaded checkpoint. This includes the precision settings, loop progress,
+        optimizer states and learning rate scheduler states.
         """
+        if not checkpoint:
+            return
```

Comment:
when can this happen?

Reply:
when Trainer(resume_from_checkpoint=None) 
(the default case)
then trainer calls restore() and we do a no-op

Reply:
shouldn't we skip calling restore at all in that case?

Reply:
no because it's actually an attempt to restore. 
this is the sequence:

1. attempt to restore from hpc_checkpoint file
2. attempt to restore from resume_from_checkpoint path if provided
3. do nothing

and could rename restore to attempt_restore if that's better?